Plugins/Public/base_plugin/PlayerBase.cpp

```diff
@@ -75,21 +73,72 @@ void PlayerBase::Spawn()
 	SyncReputationForBase();
 }
 
+bool IsVulnerabilityWindowActive(BASE_VULNERABILITY_WINDOW window, int timeOfDay)
+{
+	return ((window.start < window.end
+			&& window.start <= timeOfDay && window.end > timeOfDay)
+		|| (window.start > window.end
+			&& (window.start <= timeOfDay || window.end > timeOfDay)));
+}
+
+void PlayerBase::CheckVulnerabilityWindow(uint currTime)
+{
+	int timeOfDay = (currTime % (3600 * 24)) / 60;
+	if (IsVulnerabilityWindowActive(vulnerabilityWindow1, timeOfDay))
+	{
+		if (!vulnerableWindowStatus)
+		{
+			vulnerableWindowStatus = true;
+			siege_mode = true;
+			SyncReputationForBase();
+			shield_strength_multiplier = base_shield_strength;
+			damage_taken_since_last_threshold = 0;
+			if (shield_reinforcement_threshold_map.count(base_level))
+			{
+				base_shield_reinforcement_threshold = shield_reinforcement_threshold_map[base_level];
+			}
+			else
+			{
+				base_shield_reinforcement_threshold = FLT_MAX;
+			}
+		}
+		vulnerableWindowStatus = true;
+	}
+	else if (!single_vulnerability_window && IsVulnerabilityWindowActive(vulnerabilityWindow2, timeOfDay))
+	{
+		if (!vulnerableWindowStatus)
+		{
+			vulnerableWindowStatus = true;
+			siege_mode = true;
+			SyncReputationForBase();
```

Comment:
what about all the other stuff we do above?

Reply:
Those are all separate else-if checks, its impossible more than 1 would execute